### ISSUE TITLE
You can implement it better!

### ISSUE THREAD

**Issue by hfarazi** [neuraloperator/neuraloperator#10]
https://github.com/zongyi-li/fourier_neural_operator/blob/90bbd4c9ffa84b91ec2daa86660148861989adec/fourier_3d.py#L53

What you can do is using FFTshift:
```
def roll_n(X, axis, n):
    f_idx = tuple(slice(None, None, None) if i != axis else slice(0, n, None)
                  for i in range(X.dim()))
    b_idx = tuple(slice(None, None, None) if i != axis else slice(n, None, None)
                  for i in range(X.dim()))
    front = X[f_idx]
    back = X[b_idx]
    return torch.cat([back, front], axis)

def fftshift(data):
    for dim in range(len(data.size()) - 3, len(data.size()) - 1):
        data = roll_n(data, axis=dim, n=data.size(dim) // 2)
    return data
```
and then slice just the middle of the tensor (instead of four different corners)

**Comment by zongyi-li** [comment]
That's a very good point! Thanks for bringing this up!

**Comment by hfarazi** [comment]
I created a pull request: https://github.com/zongyi-li/fourier_neural_operator/pull/12

**Comment by StephenHogg** [comment]
@hfarazi @zongyi-li pytorch has a built-in fftshift function these days, for what it's worth

**Comment by JeanKossaifi** [comment]
Is there any advantage in doing it that way?
I had also tried to roll the coefficients but there was no speedup.

**Comment by JeanKossaifi** [comment]
Closing as inactive.
